src/Writer.php

```diff
@@ -103,13 +103,18 @@ public function writeBool($value)
         $this->writer->writeUInt8($value ? 1 : 0);
     }
 
-    public function writeVariant($value)
+    public function writeVariant($value, $type = null)
     {
-        $this->writeVariantType($value, $this->types->getTypeByValue($value));
-    }
+        if ($type === null) {
+            $type = $this->types->getTypeByValue($value);
+        }
+
+        if (is_string($type)) {
+            $this->writeType(Types::TYPE_USER_TYPE);
+
+            return $this->writeUserTypeByName($value, $type);
```

Comment:
Duh, this should not have slipped in with this PR :)
Can be ignored for now, the full functional will be introduced with the upcoming PRs